### PR TITLE
Improve organization discovery feature

### PR DESCRIPTION
**Pull request by Rashmini**
### Purpose
$subject

### Related Issues
- https://github.com/wso2/product-is/issues/16379

### Related PRs
- None

### Checklist
- [ ] e2e cypress tests locally verified.
- [ ] Manual test round performed and verified.
- [ ] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Integration tests provided. (Add links if there are any)

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: modules/i18n/src/translations/si-LK/portals/console.ts [hunk 1]
```diff
@@ -7488,60 +7491,60 @@ export const console: ConsoleNS = {
...
+                            description: "සංවිධානයේ සොයාගැනීම් ගුණාංග ලබා ගැනීමේදී දෝෂයක් ඇති විය.",
+                            message: "ලබා ගැනීම අසාර්ථකයි"
                         }
                     },
                     getEmailDomainDiscovery: {
                         error: {
-                            description: "ඊමේල් වසම් සොයාගැනීමේ වින්‍යාසය ලබා ගැනීමේදී දෝෂයක් ඇති විය",
-                            message: "මොකක්හරි වැරැද්දක් වෙලා"
+                            description: "ඊමේල් වසම් සොයාගැනීමේ වින්‍යාසය ලබා ගැනීමේදී දෝෂයක් ඇති විය.",
+                            message: "Retrieval unsuccessful"
```
Comment (thanujalk) [inline modules/i18n/src/translations/si-LK/portals/console.ts:7531]: why this in english?

File: modules/i18n/src/translations/si-LK/portals/console.ts [hunk 2]
```diff
@@ -7488,60 +7491,60 @@ export const console: ConsoleNS = {
...
                         }
                     },
                     getOrganizationListWithDiscovery: {
                         error: {
-                            description: "සොයාගැනීම් ගුණාංග සහිත සංවිධාන ලැයිස්තුව ලබා ගැනීමේදී දෝෂයක් ඇති විය",
-                            message: "මොකක්හරි වැරැද්දක් වෙලා"
+                            description: "An error occurred while getting the organization list with discovery attributes.",
```
Comment (thanujalk) [inline modules/i18n/src/translations/si-LK/portals/console.ts:7536]: why this in english?

File: modules/i18n/src/translations/si-LK/portals/console.ts (hunk 1 above)
Comment (Rashmini) [reply modules/i18n/src/translations/si-LK/portals/console.ts]: Fixed with 45fedafa3c5b995e4def060f6399037dd4c4d9fd

File: modules/i18n/src/translations/si-LK/portals/console.ts (hunk 2 above)
Comment (Rashmini) [reply modules/i18n/src/translations/si-LK/portals/console.ts]: Fixed with 45fedafa3c5b995e4def060f6399037dd4c4d9fd

File: apps/console/src/features/organization-discovery/components/add-organization-discovery-domains.tsx [hunk 3]
```diff
@@ -90,17 +91,35 @@ const FORM_ID: string = "edit-organization-email-domains-form";
...
+        isReadOnly,
+        ["data-componentid"]: componentId
+    } = props;
 
     const { t } = useTranslation();
 
     const dispatch: Dispatch = useDispatch();
 
+    const allowedScopes: string = useSelector((state: AppState) => state?.auth?.allowedScopes);
+    const featureConfig: FeatureConfigInterface = useSelector((state: AppState) => state.config.ui.features);
```
Comment (dasuni-30) [inline apps/console/src/features/organization-discovery/components/add-organization-discovery-domains.tsx:104]: Shall we add nullable checks

File: apps/console/src/features/organization-discovery/components/add-organization-discovery-domains.tsx [hunk 4]
```diff
@@ -273,17 +292,19 @@ const AddOrganizationDiscoveryDomains: FunctionComponent<AddOrganizationDiscover
...
-                                    </PrimaryButton>
+                                    !isReadOnly && !hasScopes && (
+                                        <PrimaryButton
+                                            data-componentid={ `${componentId}-form-submit-button` }
+                                            disabled={
+                                                submitting || isEmpty(emailDomains) || isEmpty(values?.organizationName)
+                                            }
+                                            loading={ submitting }
+                                            type="submit"
+                                            style={ { marginTop: "20px" } }
```
Comment (dasuni-30) [inline apps/console/src/features/organization-discovery/components/add-organization-discovery-domains.tsx:303]: Shall we avoid adding inline styles

**Review by dasuni-30 — APPROVED** [review]

File: apps/console/src/features/organization-discovery/components/add-organization-discovery-domains.tsx (hunk 4 above)
Comment (brionmario) [reply apps/console/src/features/organization-discovery/components/add-organization-discovery-domains.tsx:303]: Add this to `add-organization-discovery-domains.scss`.

File: apps/console/src/features/organization-discovery/components/add-organization-discovery-domains.tsx (hunk 3 above)
Comment (brionmario) [reply apps/console/src/features/organization-discovery/components/add-organization-discovery-domains.tsx:104]: Grab the feature specific config instead:

Ref: https://github.com/wso2/identity-apps/blob/9566504a9270c7805af809ea1087a95d53d338fb/apps/console/src/features/organization-discovery/pages/assign-organization-discovery-domains-page.tsx#L50

File: apps/console/src/features/organization-discovery/components/add-organization-discovery-domains.tsx [hunk 5]
```diff
@@ -90,17 +91,35 @@ const FORM_ID: string = "edit-organization-email-domains-form";
...
 
     const { data: discoverableOrganizations } = useGetOrganizationDiscovery(true, null, null, null);
 
     const [ emailDomains, setEmailDomains ] = useState<string[]>([]);
+    const [ hasScopes, setHasScopes ] = useState(false);
+
+    /**
+     * Set the hasScopes state based on the feature config.
+     */
+    useEffect(() => {
```
Comment (brionmario) [inline apps/console/src/features/organization-discovery/components/add-organization-discovery-domains.tsx:116]: feature config is accessed from the page too and passed down.
Can't we merge the two logics.

https://github.com/wso2/identity-apps/blob/9566504a9270c7805af809ea1087a95d53d338fb/apps/console/src/features/organization-discovery/pages/assign-organization-discovery-domains-page.tsx#L54

**Comment by Rashmini** [comment]
Will address the comments with another PR.

File: apps/console/src/features/organization-discovery/components/add-organization-discovery-domains.tsx (hunk 3 above)
Comment (Rashmini) [reply apps/console/src/features/organization-discovery/components/add-organization-discovery-domains.tsx:104]: Fixed with f14b8e1c35cbf764611d52797b803c6294e0fc0f

File: apps/console/src/features/organization-discovery/components/add-organization-discovery-domains.tsx (hunk 4 above)
Comment (Rashmini) [reply apps/console/src/features/organization-discovery/components/add-organization-discovery-domains.tsx:303]: Fixed with f14b8e1c35cbf764611d52797b803c6294e0fc0f

File: apps/console/src/features/organization-discovery/components/add-organization-discovery-domains.tsx (hunk 5 above)
Comment (Rashmini) [reply apps/console/src/features/organization-discovery/components/add-organization-discovery-domains.tsx:116]: Fixed with f14b8e1c35cbf764611d52797b803c6294e0fc0f

**Pull request merged**